FEATURE/#5: 로그인, 회원가입 페이지 (디테일 미완)

jihyun-p opened the pull request:
## *📌 Issue*
- closed #5
<!-- closed를 붙이면 해당 이슈가 PR과 함께 자동으로 close -->

## *⛳️ Work Description*
- 로그인 페이지
- 회원가입 페이지

## *📸 Screenshot*
<!-- 실행 사진이나 영상을 드래그하여 첨부해주세요. -->
<!-- <img src="이미지 주소" width=270 /> -->
![image](https://github.com/user-attachments/assets/84c3a89a-e50b-46ad-a21f-131a386c7bce)
![image](https://github.com/user-attachments/assets/c8f52577-4091-43e1-9aae-45e0afb17177)



## *📢 To Reviewers*
- **이메일에 test, 비밀번호에 1234 입력하면 로그인 가능하게 해놨습니다.** (test, 1234 입력 시 로그인 버튼 활성화)
- 시간이 부족해서 아직 디자인이나 기능적으로 부족한 부분이 있습니다.. 금방 완성시키겠습니다.
- 실수가 있는 부분이 많을 것 같은데 자유롭게 지적해주시면 감사하겠습니다!

<!-- PR 제목 형식: [FEATURE/#이슈번호] 작업 주제 -->

jihyun-p committed:
feat, style/#5: 로그인 페이지 기본 화면 구성

jihyun-p committed:
feat, style/#5: 로그인 페이지 화면 추가 구현, 페이지 이동 설정

jihyun-p committed:
feat/#5: 시작 - 로그인 페이지로 설정

dongwonyang reviewed (changes requested):
viewModel을 생성자로 넣어두셨는데 그럴 경우 hiltViewModel()은 Hilt의 DI 컨텍스트 내에서만 동작하기 때문에, 해당 Composable을 Preview에서 렌더링할 때 에러가 발생할 수 있습니다. 보통 Composable 내부에서 호출한다고 합니다.

그리고 uiState의 경우 mutableStateOf로 사용하셨는 데 uiState 같은 여러 값을 가지는 data class 를 관리할 때 Flow이므로 비동기 처리, 일부 필드만 변경 가능으로 인해 기존에 사용하던 mutableStateFlow를 기반으로  만든 후 collectAsState()로 가져오는 방식이 더 좋다고 알고 있습니다. 확인 부탁드려요.

outLinedTextFiled에서 같은 중복사용이 있을 경우 component로 빼는 것도 좋습니다. 제 pr comment에서 그쪽 방향으로 refactort 한다고 봤으니 넘어가겠습니다. 다른 화면에서도 사용할 것 같은 경우 ui/compoenent 아닐 경우 screen/login/compoenet이런식으로 패캐지 새로 만들어서 넣어주세요. 

develop과 병합해서 conflct해결은 해놓겠습니다. 확인하고시고 merge 부탁드려요.

dongwonyang committed:
Merge branch 'develop' of https://github.com/graduationProject-Crowdsourcing/front into feature/#5

dongwonyang committed:
merge develop

jihyun-p committed:
A
Merge branch 'develop' of https://github.com/graduationProject-Crowdsourcing/front into feature/#5

jihyun-p committed:
feat, design/#5: feat: 로그인/회원가입 UI 리팩토링 및 버튼 컴포넌트 적용

jihyun-p committed:
Merge remote-tracking branch 'origin/feature/#5' into feature/#5

jihyun-p committed:
feat, design/#5: 로그인 UI 정리 및 상태관리 방식 StateFlow로 전환
